spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -299,7 +300,21 @@ private Dataset<FileURI> actualFileIdentDS() {
     }
   }
 
-  private Dataset<String> listedFileDS() {
+  private Dataset<String> listWithPrefix() {
+    List<String> matchingFiles = Lists.newArrayList();
+    Iterator<org.apache.iceberg.io.FileInfo> iterator =
+        ((SupportsPrefixOperations) table.io()).listPrefix(location).iterator();
```

Comment:
Should we fall back in case `table.io()` doesn't support interface `SupportsPrefixOperations`?